### DISCUSSION TITLE
Summary of the Gordian Developer Meeting: July 5th, 2023

### DISCUSSION THREAD

**Discussion by shannona** [BlockchainCommons/Gordian-Developer-Community#114] [General & Announcements]
This was a Europe-friendly meeting (held earlier in the day). The main topics were:

## Return to URs & Animated QRs

Blockchain Commons has some new docs on using URs & Animated QRs
- [UR Overview (updated)](https://github.com/BlockchainCommons/crypto-commons/blob/master/Docs/ur-1-overview.md)
- [PSBTs for URs](https://github.com/BlockchainCommons/crypto-commons/blob/master/Docs/ur-4-psbt.md)

Blockchain Commons has also produced a video on Animated QRs
- [Animated QR Video](https://www.youtube.com/watch?v=HsFF5HPKQIk)
- Contains Sparrow & Passport
- We'd love to get more videos from other people using animated QRs

We want to spread the word about the functionality!

## UR Experiences

**Thunderbiscuit:**
- Android Dev
- Maintains language-bindings for Bitcoin Development Kit

Found UR through Hummingbird
- Exploratory
- Was looking into it for Rust or Kotlin

Might Be Able to Support Cashu:
- Heard about issue with Cashu where some tokens were too big for a single QR
   - [#31 for Nuts for Cashu](https://github.com/cashubtc/nuts/pull/31)

Feel like URs could benefit from having their own website/something, that isn't as interlinked with the rest of the Blockchain Commons work.
- ur-standard website that has *everything* you need to implement URs
- Generally, single deployment place
- Blockchain Commons is working on [an improved developer website](https://hackmd.io/RVS8ypbMShGhu2UpHeiAzw)

## Making Specs into Standards

There was encouragement that Blockchain Commons should push UR spec and especially Animated QR spec into a standard as soon as possible, before we get competing specs.

There's a bit of a tricky question about WHERE to publish them to the biggest audience, since not all are Bitcoin

What's needed most?
- The particular method for creating fountain codes, which is embedded in source code right now

## A Lifehash Use Case

Lifehash is now in Sparrow

The Use Case:
* BIP-38 Seeds, 24 words + one-word passphrase
* UX Issue: Users were creating wallets & making a typo on the passphrase
   * And that creates a new wallet!
   * Because passphrase isn't supposed to be stored anywhere, per BIP
* So: Lifehash! as you enter passphrase
   * Based on master key
   * Used master key instead of seed because might not have access to the original seed
   * It actually changes as you're typing in passphrase
   * And you should recognize it as you finish typing in passphrase
* Visual cue that's easier than just displaying the master fingerprint
   * Improves confidence in passphrase

* Great solution for BIP38, which we've found was a big vulnerability for Bitcoin loss
* Some concerns that if we use too many lifehashes, for too many different things, we'll make them less understandable for users

(Lifehash is part of the OIB, of Object Identification Block, and we'd love to see more usage of that wider bit of content.)

## IETF 117

Blockchain Commons will be at IETF 117

* Envelope Side Meeting, July 24th, 15.30-17.00, Golden Gate 4 Room
* CBOR Meeting, July 24th, 17.30-18.30, Continental 5 Room

## Self-Sovereignty

One of the foundations of Blockchain Commons; working on new articles

Self-sovereign computing is one such. It offers a new metaphor: you're on a journey on a digital sea, and you want to be the captain, not a passenger. And together we can form fleets to support each other.

## Gordian SeedTool 1.6

Demo of CSR

GST Now includes Envelopes for distributing/storing data.

Creating a seed includes metadata, such as creation date and note.
- Which more than simple hex for seed

Now can distribute a Private Key as an Envelope
- This includes type info
- Metadata such as name
- And of course the key

Because it's an Envelope you can add more info or elide old info

Similarly, you can backup Seed as a Gordian Envelope
- Includes names, creation date, notes!

Similarly, you can backup with SSKR as a Gordian Envelope
- Again, includes all the metadata, not just the seed!
- It wraps envelope of seed (and metadata), encrypts them with an ephemeral key, and then shards that

_This is the first big advantage of CSRs: we can recover metadata because it uses Envelope._

_And it's a great first look at why Envelope is crucial._

### Standardization Questions

What forms should we standardize?

BIP 38? 85? 384?

What metadata should we standardize?

329?

## Next Time

We'll be Asia-friendly times in August.

**Comment by thunderbiscuit** [comment]
Fantastic notes! 🔥

**Comment by shannona** [comment]
Thanks!
